indra/explanation/reporting.py

```diff
@@ -168,7 +168,7 @@ def stmts_from_pybel_path(path, model, from_db=True, stmts=None):
         hashes = set()
         for j in range(len(edges)):
```

Comment:
can we switch this to use the pythonic generator instead of indexing?

```python
for edge in edges:
    try:
        hashes.add(edge['annotations']['stmt_hash']
        # partOf and hasVariant edges don't have hashes
    except KeyError:
        continue
```

Reply:
wait, it seems like edges is a dictionary where keys are consecutive integers... wouldn't a list be a better data structure?

Reply:
I think it's an artifact of an earlier version when all_edges have been used directly instead of filtering to edges, it mimics the data structure of all_edges (which is I think a standard NetworkX EdgeView structure)

Reply:
I can't recall why it's implemented this way, we'll review and think about it as a follow up but I'll merge this PR first.